### PR TITLE
Fixes and tests for bidirectional mounts

### PR DESCRIPTION
**Pull request by Hoxo**

### CODE REVIEW

File: lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java [hunk 1]
```diff
@@ -64,6 +73,8 @@ protected void notifyFinished() {
             } catch (Exception e) {
                 log().error("{} Failed to start unmount dynamic disk action", logPrefix(), e);
             }
+        } else {
+            log().info("{} Mount disk finished", logPrefix());
```
Comment (imakunin) [inline lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java:77]: add disk_id (or some info about this disk), plz

File: lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java [hunk 2]
```diff
@@ -194,6 +233,23 @@ private StepResult attachVolumeToPod() {
...
+        }
+
+        log().info("{} Updating vm with new mount pod {}", logPrefix(), mountPod.podName());
+        try {
+            withRetries(log(), () ->
+                allocationContext.vmDao().setMountPod(vm.vmId(), mountPod.podName(), null));
+            vm = vm.withMountPod(mountPod.podName());
+        } catch (Exception e) {
+            log().error("{} Failed to update vm with new mount pod {}", logPrefix(), mountPod.podName(), e);
+            fail(Status.ABORTED.withDescription("Failed to update vm with new mount pod"));
```
Comment (imakunin) [inline lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java:248]: s/ABORTED/CANCELLED/

File: lzy/allocator/src/main/java/ai/lzy/allocator/alloc/impl/kuber/KuberVmAllocator.java [hunk 3]
```diff
@@ -241,19 +240,27 @@ public void unmountFromVm(Vm vm, String mountPath) throws InvalidConfigurationEx
...
+                    var returnCode = exec.exitCode().get();
+                    LOG.info("Unmount return code: {}", returnCode);
+                }
             }
         } catch (ExecutionException e) {
             LOG.warn("Unmount execution error", e);
         } catch (InterruptedException e) {
             LOG.warn("Unmount interrupted");
+        } catch (IOException e) {
+            LOG.warn("Unmount output reading error", e);
```
Comment (imakunin) [inline lzy/allocator/src/main/java/ai/lzy/allocator/alloc/impl/kuber/KuberVmAllocator.java:263]: let's add some info about vm and disk here

File: lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java [hunk 4]
```diff
@@ -164,25 +184,44 @@ private StepResult setVolumeInfo() {
...
+        if (activeMounts != null) {
+            return StepResult.ALREADY_DONE;
+        }
+
+        try {
+            activeMounts = withRetries(log(), () -> allocationContext.dynamicMountDao()
+                .getByVmAndStates(vm.vmId(), List.of(DynamicMount.State.READY), null));
+        } catch (Exception e) {
+            log().error("{} Failed to read active mounts for vm", logPrefix(), e);
+            fail(Status.ABORTED.withDescription("Failed to read active mounts for vm"));
```
Comment (imakunin) [inline lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java:197]: s/ABORTED/CANCELLED/

File: lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java (hunk 1 above)
Comment (Hoxo) [reply lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java:77]: disk_id is already in prefix, that's why I got rid of any redundant data

File: lzy/long-running/src/main/java/ai/lzy/longrunning/OperationRunnerBase.java [hunk 5]
```diff
@@ -43,52 +47,61 @@ protected OperationRunnerBase(String id, String descr, Storage storage, Operatio
 
     @Override
     protected final void execute() {
-        try {
-            if (!loadOperation()) {
-                return;
-            }
+        LogUtils.withLoggingContext(prepareLogContext(), () -> {
```
Comment (imakunin) [inline lzy/long-running/src/main/java/ai/lzy/longrunning/OperationRunnerBase.java:50]: we have the same logic in the parent `ContextAwareTask`
let's merge it

Comment (Hoxo) [reply lzy/long-running/src/main/java/ai/lzy/longrunning/OperationRunnerBase.java:50]: Merge - like how?

Comment (imakunin) [reply lzy/long-running/src/main/java/ai/lzy/longrunning/OperationRunnerBase.java:50]: we can prepare logging context in the ctor

**Review by imakunin — APPROVED** [review]

**Pull request merged**